recipe_scrapers/kitchensanctuary.py

```diff
@@ -0,0 +1,78 @@
+# mypy: allow-untyped-defs
+
+from recipe_scrapers._grouping_utils import IngredientGroup
+
+from ._abstract import AbstractScraper
+
+
+class KitchenSanctuary(AbstractScraper):
+    @classmethod
+    def host(cls):
+        return "kitchensanctuary.com"
+
+    def author(self):
+        return self.schema.author()
+
+    def title(self):
+        return self.schema.title()
+
+    def category(self):
+        return self.schema.category()
+
+    def total_time(self):
+        return self.schema.total_time()
+
+    def yields(self):
+        return self.schema.yields()
+
+    def image(self):
+        return self.schema.image()
+
+    def ingredients(self):
+        return self.schema.ingredients()
+
+    def ingredient_groups(self):
+        ingredient_groups = self.soup.find_all(
+            "div", {"class": "wprm-recipe-ingredient-group"}
+        )
+
+        results = []
+        ingredient_index_base = 0
+
+        for ingredient_group in ingredient_groups:
+
+            ingredient_count = (
+                len(ingredient_group.find_all("li")) + ingredient_index_base
+            )
+
+            purpose_heading = ingredient_group.find(
+                "h4", "wprm-recipe-ingredient-group-name"
+            )
+
+            results.append(
+                IngredientGroup(
+                    ingredients=[
+                        ingredient
+                        for ingredient in self.schema.ingredients()[
+                            ingredient_index_base:ingredient_count
+                        ]
+                    ],
+                    purpose=purpose_heading.text if purpose_heading else None,
+                )
+            )
+
+            ingredient_index_base = ingredient_count
+
+        return results
```

Comment:
```suggestion
    def ingredient_groups(self):
        return group_ingredients(
            self.ingredients(),
            self.soup,
            ".wprm-recipe-ingredient-group-name",
            ".wprm-recipe-ingredient-group li",
        )
```
Is there any reason not to use the `ingredient_groups` helper function in place of the custom logic here?

Reply:
This change would also require a small change to the test asserts as well

Reply:
HI @jknndy, thanks for the feedback. Although I didn't have reason initially to use this helper function, testing this change seems to break my test. This is after changing some things in the file. This change doesn't seem to break the ingredient_group test but the ingredients test itself. The test seems to get confused in the ingredients list because there are some repeated ingredients e.g. "1 tsp salt" & "1/2 tsp salt" and would make weird suggestions in the diff of the unit test. This doesn't happen with the custom method. So what changes to the test asserts did you have in mind because they may fix this issue?

Reply:
interesting...i see the issue with the garlic salt line too. 

This isn't any reason to not merge this pull, i'll take another look through the code and open an issue about this.